app/helpers/application_helper/button/middleware_domain_server_action.rb

```diff
@@ -8,6 +7,6 @@ def visible?
   private
 
   def in_domain?
-    @record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?)
+    !@record.nil? && (@record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?))
```

Comment:
I removed because these buttons needs to be displayed on some screens that doesn't have an associated  a `@record`

This line does a test to see if the server is in domain but only for the cases when the `@record` exists, that is why I verified first if `@record`  is not nil. I'll do the change to `present?` if that fits better in this.